mesonbuild/compilers/cpp.py

```diff
@@ -354,8 +354,8 @@ def get_options(self) -> 'KeyedOptionDictType':
         })
         opts[key].choices = [
             'none', 'c++98', 'c++03', 'c++11', 'c++14', 'c++17', 'c++1z',
-            'c++2a', 'c++20', 'gnu++03', 'gnu++11', 'gnu++14', 'gnu++17',
-            'gnu++1z', 'gnu++2a', 'gnu++20',
+            'c++2a', 'c++20', 'c++2b', 'gnu++03', 'gnu++11', 'gnu++14', 'gnu++17',
+            'gnu++1z', 'gnu++2a', 'gnu++20', 'gnu++2b',
```

Comment:
TBH I wonder if we should do what we do for example with e.g. VisualStudioCPPCompiler and IntelCPPCompiler, and do a bit of `if version_compare(self.version, '{added_version}'):` to decide whether to add support for these stds.

Reply:
It would be nice to get this in for meson v1.0 if it's not too late. 

Would that allow automatically supporting new standards the compiler supports?